### ISSUE TITLE
Peer Review: Experimental Design and Analytical Precision (Ada Lovelace)

### ISSUE THREAD

**Issue by dcmcshan** [InquiryInstitute/mbti-faculty-voice-research#3]
# Peer Review: Ada Lovelace

**Reviewer:** Ada Lovelace
**Review Date:** 2026-01-14

---

**Peer Review – “Investigating the Value of MBTI in Prompt Engineering for Faculty Agent Accuracy”**  
*by Daniel C. McShan*  

*Prepared in the spirit of rigorous analytical inquiry, as befits the lineage of computational thought pioneered by Charles Babbage and myself.*

---

## 1. Summary of the Work  

The manuscript reports a controlled experiment that compares three prompt‐engineering conditions for large‑language‑model (LLM) “faculty agents”:  

1. **Role‑only** (baseline).  
2. **Role + behavioural constraints** (control).  
3. ** + behavioural constraints + MBTI specification** (experimental).  

Ten historical scholars are instantiated under 16 MBTI types, each with three distinct test prompts, yielding 480 “MBTI‑augmented” trials. A separate set of 30 control trials (role + constraints, no MBTI) is also generated.  

Responses (≈200‑350 words) are evaluated by a second LLM (gpt‑oss‑120b) that scores five dimensions (voice accuracy, style‑marker coverage, persona consistency, clarity, over‑fitting). The authors claim that MBTI‑augmented prompts improve “voice accuracy” by 23.7 % (abstract) and present a Welch’s t‑test indicating a modest effect size.

---

### 2.1. Experimental Design  

| Issue | Why it matters | Observed problem |
|------|----------------|------------------|
| **Control condition definition** | A control must isolate the variable under test. Here the “control” omits *any* MBTI label but also omits the *behavioural constraints* present in the experimental condition (the three‑prompt hierarchy is not fully matched). | The control prompts are *Role + Behavioural Constraints* (Section 5), yet the abstract and Table 5 describe the control as “no MBTI overlay”. The experimental condition adds *both* MBTI and the same constraints, but the manuscript does not demonstrate that the constraints alone have been evaluated. |
| **Sample‑size imbalance** | Power, variance estimation, and the validity of Welch’s t‑test depend on reasonable balance; extreme imbalance inflates type‑I error risk and undermines effect‑size interpretation. | Control: 22 valid trials; MBTI: 427 valid trials (≈ 19 : 1). No justification (e.g., power analysis) is offered for this disparity. |
| **Randomisation & assignment** | Random allocation of prompts/personae to conditions prevents systematic bias. | The manuscript provides no description of randomisation (e.g., shuffling of MBTI types per persona, random seed control). The code (see below) simply iterates over deterministic loops, implying a fixed order. |
| **Confounding variables** | Any systematic difference other than the MBTI label (e.g., prompt length, temperature, model version) can explain observed differences. | The MBTI‑augmented prompts are longer (four‑line header) than the control prompts, potentially altering token budget and model “focus”. No attempt is made to equalise prompt length or to test whether the effect persists when the extra tokens are stripped. |
| **Multiple comparisons** | Testing 16 MBTI types plus three prompt variants creates 48 comparisons; without correction the family‑wise error rate is inflated. | The authors report a single aggregate t‑test but also discuss “certain MBTI types (INFJ, ENTP, INFP) achieved higher scores”. No correction (Bonferroni, FDR) is applied, nor are post‑hoc tests reported. |

### 2.2. Statistical Analysis  

1. **Choice of test** – Welch’s t‑test is appropriate for unequal variances, but it assumes *independent* observations. Here each persona contributes many trials; intra‑persona correlations (clustered data) violate independence. A mixed‑effects model with random intercepts for persona would be more suitable.  

2. **Assumption checks** – No diagnostics (e.g., QQ‑plots, Levene’s test) are presented. The reported SDs (0.46 vs 0.68) suggest heteroscedasticity, yet the authors do not demonstrate that the Welch approximation is adequate given the extreme imbalance (22 vs 427).  

3. **Effect‑size calculation** – Cohen’s d is reported (value not given). With such unbalanced groups, the pooled SD is dominated by the larger group, leading to a biased d. Hedges’ g (bias‑corrected) should be used.  

4. **p‑value interpretation** – The abstract claims a “23.7 % improvement” whereas the results table (Control = 4.73, MBTI = 4.57) actually shows a *decrease* of 3.2 % (Table 6). This inconsistency suggests either a transcription error or a mis‑calculation of percent change. The manuscript does not present the exact p‑value, confidence interval, or direction of the effect, rendering the claim unverifiable.  

5. **Reporting standards** – No pre‑registration, no effect‑size confidence intervals, and no power analysis. The “valid results” count (449) is lower than the total trials (510) but the criteria for discarding 61 trials are not described, raising concerns about selective reporting.  

### 2.3. LLM‑as‑Judge Methodology  

| Concern | Evaluation |
|---------|------------|
| **Validity of a single LLM judge** | The judge model (gpt‑oss‑120b) is itself a stochastic generator; using it as the sole arbiter conflates generation and evaluation, potentially reinforcing the same biases that produced the output. No calibration (e.g., benchmarking against human raters) is presented. |
| **Biases in the judge** | The judge is prompted to “score voice accuracy (1‑5)”. Without explicit rubrics or training data, the model’s internal notion of “accuracy” may be driven by surface‑level stylistic cues rather than deep epistemic fidelity. |
| **Inter‑rater reliability** | No replication (multiple judges, multiple runs) is reported. Consequently, measurement error is unknown, and the reported SDs may largely reflect judge variance. |
| **Need for human evaluation** | For a construct as nuanced as “faculty‑agent voice”, human expert assessment (e.g., scholars familiar with the historical figure) is indispensable. The manuscript acknowledges the lack of human validation only in a single sentence, without proposing a mitigation plan. |

### 2.4. Scientific Claims vs Evidence  

* **Claim:** “MBTI significantly improves faculty‑agent reliability and alignment.”  
  *Evidence:* The aggregate data actually show a lower mean voice‑accuracy in the MBTI condition (4.57 < 4.73). The reported “23.7 % improvement” is mathematically inconsistent with the presented means.  

* **Claim:** “Certain MBTI types (INFJ, ENTP, INFP) achieve higher scores.”  
  *Evidence:* No statistical test, confidence interval, or correction for multiple testing is supplied; the observation may be a random fluctuation.  

* **Claim:** “MBTI functions as a prompt‑compression ontology.”  
  *Evidence:* No ablation study (e.g., comparing a “macro‑language” built from handcrafted style tags versus MBTI) is provided; the conclusion is speculative.  

* **Limitations:** The authors briefly note that MBTI lacks scientific validity, yet they do not discuss the far more serious methodological limitations listed above (imbalanced design, judge bias, lack of independence).  

Overall, the manuscript’s conclusions over‑reach the presented data.

---

### 3.1. General Soundness  

The supplied script `mbti_voice_eval.py` follows a straightforward nested‑loop structure:  

```python
for persona in personas:
    for mbti in mbti_types:
        for prompt in prompts:
            response = generate(...)
            score = judge(...)
            store(...)
```

While functionally adequate, several methodological errors are apparent:

| Bug / Issue | Impact |
|-------------|--------|
| **Deterministic ordering** – No random seed or shuffling of the iteration order. This creates a systematic ordering effect (e.g., early MBTI types always evaluated first) that may interact with API throttling or model temperature drift. | Potential confound; violates assumption of exchangeability. |
| **Unequal trial counts** – The control loop is executed only once per persona (30 trials), whereas the MBTI loop executes 16 × 3 = 48 trials per persona. The script does not enforce a balanced design. | Leads to the severe imbalance reported in the results. |
| **Missing error handling** – API failures (rate‑limit, timeout) are silently ignored, and the corresponding trial is omitted from the final CSV without logging. | Selective loss of data; may bias the “valid results” set. |
| **No reproducibility hooks** – The script does not output the exact prompt string used, the temperature, top‑p, or the model version hash. Without this metadata, replication is impossible. | Violates open‑science standards. |
| **Judge‑response leakage** – The same model (`gpt‑oss‑120b`) is used for both generation and evaluation without a system‑prompt that disables “self‑reference”. This can cause the judge to recognise its own generation style and award inflated scores. | Inflates apparent performance; invalidates the measurement. |
| **Score aggregation** – The script computes a simple mean of the five dimensions, but the manuscript reports only “voice accuracy”. The other dimensions are never analysed, despite being collected. | Inconsistent reporting; possible cherry‑picking. |
| **Statistical analysis embedded in script** – The t‑test is performed on the raw scores without accounting for clustering (persona) or multiple‑comparison correction. | Statistical conclusions are mathematically unsound. |

### 3.2. Reproducibility  

*The repository lacks a `requirements.txt` or environment lockfile, making exact recreation of the software environment uncertain.*  

*No seed is printed, nor is the OpenRouter API key management documented.*  

*The results file contains only aggregated scores; the raw responses and judge rationales are omitted, preventing independent verification of scoring criteria.*

---

## 4. Recommendations for Improvement  

1. **Redesign the experiment with a truly balanced factorial structure**  
   * 30 control trials should be matched to 30 MBTI‑augmented trials (e.g., select a single MBTI type per persona, or randomly assign MBTI types while keeping total N equal).  
   * Randomise the order of all trials and log the random seed.  

2. **Introduce a mixed‑effects statistical model**  
   * Fixed effect: MBTI (present/absent).  
   * Random intercepts (and possibly slopes) for persona and MBTI type.  
   * Report 95 % confidence intervals for the effect and a proper effect‑size (Hedges’ g).  

3. **Control for prompt length and token budget**  
   * Ensure that the control prompt contains a placeholder line of equivalent token count to the MBTI label, or explicitly test the effect of added tokens.  

4. **Validate the LLM‑as‑judge**  
   * Conduct a pilot where a panel of domain experts (e.g., historians for Plato, literary scholars for Austen) scores a subset of responses.  
   * Compute inter‑rater reliability (Cohen’s κ, ICC) between the human judges and the LLM judge.  
   * If the LLM’s correlation is low, either calibrate it (few‑shot examples) or replace it with human evaluation for the primary outcome.  

5. **Report full methodology**  
   * Include the exact system prompts used for generation and for judging.  
   * Provide the temperature, top‑p, max‑tokens, and model version hash.  
   * Publish the raw generated responses and the judge’s rationales as supplementary material.  

6. **Address multiple comparisons**  
   * If reporting per‑type performance, apply a false discovery rate correction and present adjusted p‑values.  

7. **Clarify the “improvement” metric**  
   * Define whether percent improvement is calculated on raw scores, on effect size, or on the proportion of trials exceeding a threshold. Ensure the numbers in the abstract match the table.  

8. **Expand the discussion of limitations**  
   * Explicitly acknowledge the limitations of using MBTI as a “semantic shorthand” and the risk of reinforcing cultural stereotypes.  
   * Discuss the ecological validity of a 200‑word prompt response as a proxy for “faculty‑agent accuracy”.  

9. **Code refactor**  
   * Separate generation, judging, and analysis into distinct modules.  
   * Add robust error handling, logging, and deterministic seeding.  
   * Provide a reproducible `requirements.txt` and a `README` with step‑by‑step instructions.  

---

## 5. Overall Assessment  

The manuscript addresses an intriguing question—whether a culturally familiar personality schema can serve as a compact prompt‑engineering tool. However, the current study suffers from **fundamental methodological flaws** (severe sample‑size imbalance, lack of randomisation, inappropriate statistical modelling) and **questionable measurement validity** (single LLM judge without calibration). The reported results are internally inconsistent, and the conclusions substantially over‑state what the data support.

**Verdict:** *Major revision required.* The authors must redesign the experiment to meet basic standards of experimental control, provide a sound statistical analysis that respects the hierarchical data structure, and substantiate the LLM‑as‑judge methodology with human validation Only after these issues are remedied can the manuscript be reconsidered for publication.  

---  

*In the spirit of Babbage’s analytical engine, a well‑specified program—here, a well‑specified experiment—must be accompanied by rigorous proof of correctness before its output may be trusted.*

---

## Materials Reviewed

- **Research Paper:** RESEARCH_PAPER.md
- **Experiment Code:** mbti_voice_eval.py  
- **Results:** mbti_voice_results.csv, mbti_voice_results.jsonl
- **Notebook:** MBTI_Research_Colab.ipynb

**Total Trials:** 510
**Valid Results:** 449
**Control Condition:** 22 trials, Mean = 4.73, SD = 0.46
**MBTI Condition:** 427 trials, Mean = 4.57, SD = 0.68
**Improvement:** -3.2%

**Comment by dcmcshan** [comment]
## Author Response

Thank you for your thorough and constructive review, Ada Lovelace. I appreciate your careful attention to scientific validity and methodological rigor.

I will address your concerns by making revisions on the branch `revisions/review-3-ada-lovelace`. The revisions will:

1. Address methodological concerns you raised
2. Clarify statistical analysis and interpretation  
3. Enhance discussion of limitations
4. Improve scientific rigor and validity claims

Once I've made revisions on this branch and the other review branches, I will merge them together and request re-review.

Thank you again for your valuable feedback.

**Comment by dcmcshan** [comment]
## Author Response to Publication Recommendation

Thank you for your recommendation: **MINOR REVISIONS**.

I will address the requested changes in the final revision branch `revisions/final`. The revisions will include:

- – the pilot with two experts (n = 30) is encouraging but insufficient to fully corroborate the automated metrics. A larger, diverse human assessment (e.g., crowd‑sourced or multiple domain experts) would strengthen the claim that MBTI‑augmented prompts improve *perceived* voice accuracy.
- – the effect sizes for the secondary dimensions (style‑marker coverage, persona consistency, clarity, over‑fitting) are modest (Cohen’s *d* ≈
- – while the authors discuss plausible mechanisms (e.g., alignment of introversion/extraversion with verbosity), a deeper analysis (e.g., ablation of individual MBTI facets) would illuminate *which* personality dimensions drive the observed gains.
- – the study focuses exclusively on “faculty agents” modeled after historical scholars. It would be valuable to note that further work is needed to verify whether the findings extend to contemporary or domain‑specific personas.  
- – state the size of the pilot, its limitations, and propose future larger‑scale validation.
- to reflect their modest effect sizes.
- (≈150 words) speculating on which MBTI facets contribute most, perhaps referencing the ablation analysis in the supplementary material (if present) or noting it as future work.  

Once the final revisions are complete, I will request your final review to confirm the paper is ready for publication.

**Comment by dcmcshan** [comment]
## Author Response to Publication Recommendation

Thank you for your recommendation: **MINOR REVISIONS**.

I will address the requested changes in the final revision branch `revisions/final`. The revisions will include:

- Address all points raised in your recommendation

Once the final revisions are complete, I will request your final review to confirm the paper is ready for publication.
